targets/lenovo-x1-carbon.nix

```diff
@@ -0,0 +1,181 @@
+
+              virtualization.microvm-host.enable = true;
+              host.networking.enable = true;
+              virtualization.microvm.netvm = {
+                enable = true;
+                extraModules = netvmExtraModules;
+              };
+              virtualization.microvm.guivm = {
+                enable = true;
+                extraModules = guivmExtraModules;
+              };
+              virtualization.microvm.appvm = {
+                enable = true;
+                vms = [
+                  {
+                    name = "chromium";
+                    packages = [pkgs.chromium];
+                    ipAddress = "192.168.101.5/24";
+                    macAddress = "02:00:00:03:03:05";
+                    ramMb = 3072;
+                    cores = 4;
+                  }
+                  {
+                    name = "gala";
+                    packages = [pkgs.gala-app];
+                    ipAddress = "192.168.101.6/24";
+                    macAddress = "02:00:00:03:03:06";
+                    ramMb = 1536;
+                    cores = 2;
+                  }
+                  {
+                    name = "zathura";
+                    packages = [pkgs.zathura];
+                    ipAddress = "192.168.101.7/24";
+                    macAddress = "02:00:00:03:03:07";
+                    ramMb = 512;
+                    cores = 1;
+                  }
+                ];
+                extraModules = [{}];
+              };
+
+              # Enable all the default UI applications
+              profiles = {
+                applications.enable = false;
+                #TODO clean this up when the microvm is updated to latest
+                release.enable = variant == "release";
+                debug.enable = variant == "debug";
+              };
+              windows-launcher.enable = false;
+            };
+          })
+
+          formatModule
+
+          #TODO: how to handle the majority of laptops that need a little
+          # something extra?
+          # SEE: https://github.com/NixOS/nixos-hardware/blob/master/flake.nix
+          # nixos-hardware.nixosModules.lenovo-thinkpad-x1-10th-gen
+
+          {
+            boot.kernelParams = [
+              "intel_iommu=on,igx_off,sm_on"
+              "iommu=pt"
+
+              # Passthrough Intel WiFi card 8086:51f1
+              # Passthrough Intel Iris GPU 8086:a7a1
+              "vfio-pci.ids=8086:51f1,8086:a7a1"
+            ];
+          }
+        ]
+        ++ (import ../modules/module-list.nix)
+        ++ extraModules;
+    };
+  in {
+    inherit hostConfiguration;
+    name = "${name}-${variant}";
+    package = hostConfiguration.config.system.build.${hostConfiguration.config.formatAttr};
+  };
+  debugModules = [../modules/development/usb-serial.nix {ghaf.development.usb-serial.enable = true;}];
+  targets = [
+    (generic-x86 "debug" debugModules)
+    (generic-x86 "release" [])
```

Comment:
is it still generic-x86 or should it be x1?

Reply:
Fixed